### ISSUE TITLE
Processing time difference between legacy functions and new classes

### ISSUE THREAD

**Issue by adbar** [adbar/simplemma#107]
Even if the existing/legacy functions still work there is a x2.5 time difference between them and the new approach based on your classes @juanjoDiaz. See below for a graphical explanation by pyinstrument (the files are in HTML format, GitHub wouldn't take them without the additional extension).

A lot of time seems to go to classes init each time the lemmatization function is run. If it's the culprit there is probably a way to avoid re-initializing everything each time, what do you think?

[pyinstrument-legacy.html.log](https://github.com/adbar/simplemma/files/11755811/pyinstrument-legacy.html.log)
[pyinstrument-new.html.log](https://github.com/adbar/simplemma/files/11755812/pyinstrument-new.html.log)

**Comment by juanjoDiaz** [comment]
Indeed. It seems that initializing everything takes its toll.
We could pre-initialize at module level a greedy and a non-greedy `DefaultStrategy` using the same `DictionaryFactory` and them choose between the two in the function so it's not re-initialized every time.

Downside is that whatever the `DictionaryFactory` caches will stay there while the script is up (the user can't control) but that's are better because they offer more control to the user...

**Comment by adbar** [comment]
Yes the pre-initialization makes perfect sense. The users keeping the legacy functions won't bother what the class does and the ones into customization will hopefully use the classes so it should be fine.

**Comment by adbar** [comment]
@juanjoDiaz Could you please draft a PR to address this issue?
